File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] folder) {
        Arrays.sort(folder);
        List<String> ans = new ArrayList<>();

        for (String f : folder) {
            if (ans.isEmpty() || !f.startsWith(ans.get(ans.size() - 1) + "/")) {
                ans.add(f);
            }
        }
        return ans;
    }
}

File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        sort(folder.begin(), folder.end());
        vector<string> res = {folder[0]};
        for (int i = 1; i < folder.size(); i++) {
            int a = res.back().size(), b = folder[i].size();
            if (a >= b || res.back() != folder[i].substr(0, a) || folder[i][a] != '/')
                res.push_back(folder[i]);
        }
        return res;
    }
};
